runtime/centrifuge/src/migrations.rs

```diff
@@ -203,4 +212,135 @@ mod asset_registry {
 			Ok(())
 		}
 	}
+
+	/// Migrate all the registered asset's metadata to the new version of
+	/// `CustomMetadata` which contains a `AssetTransferability` property. At
+	/// this point in time, the `transferability` of Tranche tokens should be
+	/// set to `AssetTransferability::Xcm` and for all other tokens to
+	/// `AssetTransferability::Xcm`, with the exception of `Currency::Staking`
+	/// tokens which are not registered in the first place.
+	pub struct AssetTransferabilityMigration;
+
+	// The old orml_asset_registry Metadata storage using v0::CustomMetadata
+	#[storage_alias]
+	type Metadata<T: orml_asset_registry::Config> = StorageMap<
+		orml_asset_registry::Pallet<T>,
+		Twox64Concat,
+		CurrencyId,
+		AssetMetadata<Balance, v0::CustomMetadata>,
+		OptionQuery,
+	>;
+
+	impl OnRuntimeUpgrade for AssetTransferabilityMigration {
+		fn on_runtime_upgrade() -> Weight {
+			orml_asset_registry::Metadata::<Runtime>::translate(
+				|asset_id: CurrencyId, old_metadata: AssetMetadata<Balance, v0::CustomMetadata>| {
+					match asset_id {
+						CurrencyId::Staking(_) => None,
+						CurrencyId::Tranche(_, _) => Some(to_metadata_v1(
+							old_metadata,
+							v1::CrossChainTransferability::Connectors,
+						)),
+						_ => Some(to_metadata_v1(
+							old_metadata,
+							v1::CrossChainTransferability::Xcm,
+						)),
+					}
+				},
+			);
+
+			Weight::from_ref_time(200_000_000)
```

Comment:
Eventually, we should return the weight based on the number of read and written items, e.g. like

```suggestion
			let n = orml_asset_registry::Metadata::<Runtime>::iter().count();
			T::DbWeight::get().reads_writes(n, n)
```

Reply:
Thanks William!